### DISCUSSION TITLE
Not able to build the operator, even the image

### DISCUSSION THREAD

**Discussion by crombus** [keycloak/keycloak#23257] [Archive]
[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR] OptionValidationTest.failMissingOptionValue(LaunchResult) » Bootstrap Failed to create the application model for org.keycloak:keycloak-quarkus-integration-tests::jar:999.0.0-SNAPSHOTnull
[ERROR] OptionValidationTest.failMultipleOptionValue(LaunchResult) » Bootstrap Failed to create the application model for org.keycloak:keycloak-quarkus-integration-tests::jar:999.0.0-SNAPSHOTnull
[ERROR] OptionValidationTest.failUnknownOption(LaunchResult) » Bootstrap Failed to create the application model for org.keycloak:keycloak-quarkus-integration-tests::jar:999.0.0-SNAPSHOTnull
[ERROR] OptionValidationTest.failUnknownOptionEqualsSeparatorNotShowingValue(LaunchResult) » Bootstrap Failed to create the application model for org.keycloak:keycloak-quarkus-integration-tests::jar:999.0.0-SNAPSHOTnull
[ERROR] OptionValidationTest.failUnknownOptionWhitespaceSeparatorNotShowingValue(LaunchResult) » Bootstrap Failed to create the application model for org.keycloak:keycloak-quarkus-integration-tests::jar:999.0.0-SNAPSHOTnull
[ERROR] OptionValidationTest.failWithFirstOptionOnMultipleUnknownOptions(LaunchResult) » Bootstrap Failed to create the application model for org.keycloak:keycloak-quarkus-integration-tests::jar:999.0.0-SNAPSHOTnull
[ERROR] ShowConfigCommandTest.testShowConfigCommandHidesCredentialsInProfiles(LaunchResult) » Bootstrap Failed to create the application model for org.keycloak:keycloak-quarkus-integration-tests::jar:999.0.0-SNAPSHOTnull
[ERROR] ShowConfigCommandTest.testShowConfigCommandShowsRuntimeConfig(LaunchResult) » Bootstrap Failed to create the application model for org.keycloak:keycloak-quarkus-integration-tests::jar:999.0.0-SNAPSHOTnull
[ERROR] ShowConfigCommandTest.testShowConfigCommandWithAllShowsAllProfiles(LaunchResult) » Bootstrap Failed to create the application model for org.keycloak:keycloak-quarkus-integration-tests::jar:999.0.0-SNAPSHOTnull
[ERROR] FipsDistTest.testFipsNonApprovedMode:38->runOnFipsEnabledDistribution:143->lambda$testFipsNonApprovedMode$0:39 » Runtime Failed to start the server
[INFO]
[ERROR] Tests run: 184, Failures: 0, Errors: 10, Skipped: 11
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Keycloak 999.0.0-SNAPSHOT:
[INFO]
[INFO] Keycloak BOM Parent ................................ SUCCESS [06:10 min]
[INFO] Keycloak BOM for adapters .......................... SUCCESS [ 0.061 s]
[INFO] Keycloak BOM for server extensions ................. SUCCESS [ 0.052 s]
[INFO] Keycloak BOM utilities for the quickstarts ......... SUCCESS [ 0.046 s]
[INFO] Keycloak ........................................... SUCCESS [ 0.058 s]
[INFO] Keycloak Common .................................... SUCCESS [18:10 min]
[INFO] Keycloak Core ...................................... SUCCESS [ 11.427 s]
[INFO] Keycloak Server SPI ................................ SUCCESS [ 32.549 s]
[INFO] Keycloak Server Private SPI ........................ SUCCESS [ 11.160 s]
[INFO] Keycloak Crypto Parent ............................. SUCCESS [ 0.032 s]
[INFO] Keycloak Crypto Default ............................ SUCCESS [ 11.178 s]
[INFO] Keycloak SAML Core Public API ...................... SUCCESS [ 31.511 s]
[INFO] Keycloak SAML Core ................................. SUCCESS [ 7.820 s]
[INFO] Keycloak REST Services ............................. SUCCESS [01:07 min]
[INFO] Keycloak Crypto FIPS 140-2 Integration ............. SUCCESS [ 15.079 s]
[INFO] Keycloak Crypto Wildfly Elytron Integration ........ SUCCESS [01:03 min]
[INFO] Keycloak JavaScript Adapter (JAR) .................. SUCCESS [02:14 min]
[INFO] Keycloak Themes .................................... SUCCESS [02:02 min]
[INFO] Keycloak Dependencies Parent ....................... SUCCESS [ 0.020 s]
[INFO] Keycloak Dependencies Server Min ................... SUCCESS [ 0.035 s]
[INFO] Keycloak Model Parent .............................. SUCCESS [ 0.017 s]
[INFO] Keycloak Legacy Database Support ................... SUCCESS [ 0.638 s]
[INFO] Keycloak Legacy Database Support - Private classes . SUCCESS [ 2.386 s]
[INFO] Keycloak Model JPA ................................. SUCCESS [ 57.070 s]
[INFO] Keycloak Model Java Annotations and Processor ...... SUCCESS [ 2.233 s]
[INFO] Keycloak Model Map ................................. SUCCESS [ 30.032 s]
[INFO] Keycloak Model Map JPA ............................. SUCCESS [ 27.659 s]
[INFO] Keycloak Model Map LDAP ............................ SUCCESS [ 2.096 s]
[INFO] Keycloak Kerberos Federation ....................... SUCCESS [ 0.607 s]
[INFO] Keycloak LDAP UserStoreProvider .................... SUCCESS [ 5.473 s]
[INFO] Keycloak Legacy Database Support - REST services ... SUCCESS [ 0.903 s]
[INFO] Keycloak Model Infinispan .......................... SUCCESS [ 57.121 s]
[INFO] Keycloak Model Hot Rod ............................. SUCCESS [ 20.306 s]
[INFO] Keycloak Model Map File ............................ SUCCESS [ 2.840 s]
[INFO] Keycloak SSSD Federation ........................... SUCCESS [ 40.328 s]
[INFO] Keycloak Authz: Parent ............................. SUCCESS [ 0.048 s]
[INFO] Keycloak AuthZ: Policy Provider Parent ............. SUCCESS [ 0.056 s]
[INFO] Keycloak AuthZ: Common Policy Providers ............ SUCCESS [ 1.107 s]
[INFO] Keycloak Dependencies Server All ................... SUCCESS [ 0.125 s]
[INFO] Keycloak Quarkus Parent ............................ SUCCESS [ 0.034 s]
[INFO] Keycloak Configuration API ......................... SUCCESS [ 0.863 s]
[INFO] Keycloak JavaScript Parent ......................... SUCCESS [ 6.907 s]
[INFO] Keycloak Account UI ................................ SUCCESS [03:13 min]
[INFO] Keycloak Admin UI .................................. SUCCESS [02:25 min]
[INFO] Keycloak Administration UI ......................... SUCCESS [ 0.098 s]
[INFO] Admin UI REST extensions ........................... SUCCESS [02:16 min]
[INFO] Keycloak Quarkus Server Extension .................. SUCCESS [08:24 min]
[INFO] Keycloak Quarkus Server Deployment ................. SUCCESS [02:53 min]
[INFO] Keycloak Docs Parent ............................... SUCCESS [ 0.015 s]
[INFO] Keycloak Guides Maven Plugin ....................... SUCCESS [01:24 min]
[INFO] Keycloak Guides .................................... SUCCESS [02:45 min]
[INFO] Keycloak Federation ................................ SUCCESS [ 0.018 s]
[INFO] Keycloak Util Embedded LDAP ........................ SUCCESS [ 53.641 s]
[INFO] Keycloak Util Parent ............................... SUCCESS [ 0.019 s]
[INFO] Keycloak Integration ............................... SUCCESS [ 0.020 s]
[INFO] Keycloak Admin REST Client JavaEE .................. SUCCESS [ 17.039 s]
[INFO] Keycloak Admin REST Client ......................... SUCCESS [ 29.899 s]
[INFO] Keycloak Client Registration API ................... SUCCESS [ 0.255 s]
[INFO] Keycloak Client CLI ................................ SUCCESS [ 0.021 s]
[INFO] Keycloak Client Registration CLI ................... SUCCESS [ 3.939 s]
[INFO] Keycloak Admin CLI ................................. SUCCESS [ 4.422 s]
[INFO] Keycloak Client CLI Distribution ................... SUCCESS [ 0.538 s]
[INFO] Keycloak Adapter SPI ............................... SUCCESS [ 0.299 s]
[INFO] Keycloak Tomcat Adapter SPI ........................ SUCCESS [ 11.183 s]
[INFO] Keycloak Undertow Integration SPI .................. SUCCESS [ 8.473 s]
[INFO] Keycloak Servlet Integration ....................... SUCCESS [ 0.263 s]
[INFO] Keycloak Jakarta Servlet Integration ............... SUCCESS [ 0.290 s]
[INFO] Common JBoss/Wildfly Core Classes .................. SUCCESS [ 2.896 s]
[INFO] Keycloak Jetty Adapter SPI ......................... SUCCESS [ 8.434 s]
[INFO] Keycloak Client Adapter SPI Modules ................ SUCCESS [ 0.017 s]
[INFO] Keycloak SAML Client Adapter Public API ............ SUCCESS [ 0.174 s]
[INFO] Keycloak SAML Client Adapter Core .................. SUCCESS [ 2.988 s]
[INFO] Keycloak Jetty Core SAML Integration ............... SUCCESS [ 0.335 s]
[INFO] Keycloak Jetty 9.4.x SAML Integration .............. SUCCESS [ 5.810 s]
[INFO] Keycloak SAML Jetty Integration .................... SUCCESS [ 0.014 s]
[INFO] Keycloak Undertow SAML Adapter ..................... SUCCESS [ 0.245 s]
[INFO] Keycloak SAML Tomcat Integration ................... SUCCESS [ 0.018 s]
[INFO] Keycloak Tomcat Core SAML Integration .............. SUCCESS [ 0.216 s]
[INFO] Keycloak Tomcat SAML Integration ................... SUCCESS [ 0.201 s]
[INFO] Keycloak Authz: Client API ......................... SUCCESS [ 0.408 s]
[INFO] Keycloak Authz: Policy Enforcer .................... SUCCESS [ 4.031 s]
[INFO] Keycloak Adapter Core .............................. SUCCESS [ 2.407 s]
[INFO] Keycloak WildFly Elytron SAML Adapter .............. SUCCESS [ 7.263 s]
[INFO] Keycloak Wildfly SAML Adapter Subsystem ............ SUCCESS [ 55.933 s]
[INFO] Keycloak WildFly Elytron Jakarta SAML Adapter ...... SUCCESS [ 1.038 s]
[INFO] Keycloak Wildfly Jakarta SAML Adapter Subsystem .... SUCCESS [ 4.004 s]
[INFO] Keycloak SAML Wildfly Integration .................. SUCCESS [ 0.016 s]
[INFO] Keycloak SAML Servlet Filter ....................... SUCCESS [ 0.296 s]
[INFO] Keycloak SAML Jakarta Servlet Filter ............... SUCCESS [ 0.395 s]
[INFO] Keycloak SAML Client Adapter Modules ............... SUCCESS [ 0.016 s]
[INFO] Keycloak Installed Application ..................... SUCCESS [ 0.363 s]
[INFO] Keycloak JAX-RS OAuth Client ....................... SUCCESS [ 0.263 s]
[INFO] Keycloak Jetty Core Integration .................... SUCCESS [ 0.357 s]
[INFO] Keycloak Jetty 9.4.x Integration ................... SUCCESS [ 0.344 s]
[INFO] Keycloak Jetty Integration ......................... SUCCESS [ 0.021 s]
[INFO] Keycloak Servlet Filter Adapter Integration ........ SUCCESS [ 0.333 s]
[INFO] Keycloak Servlet Filter Adapter Integration ........ SUCCESS [ 0.883 s]
[INFO] Keycloak Tomcat Integration ........................ SUCCESS [ 0.031 s]
[INFO] Keycloak Tomcat Core Integration ................... SUCCESS [ 0.451 s]
[INFO] Keycloak Tomcat Integration ........................ SUCCESS [ 0.444 s]
[INFO] Keycloak Undertow Integration ...................... SUCCESS [ 0.840 s]
[INFO] spring-boot-container-bundle ....................... SUCCESS [ 0.349 s]
[INFO] Keycloak Spring Security Integration ............... SUCCESS [ 15.785 s]
[INFO] Keycloak Spring Boot Adapter Core .................. SUCCESS [ 44.243 s]
[INFO] Keycloak Spring Boot 2 Integration ................. SUCCESS [ 6.028 s]
[INFO] Keycloak Wildfly Elytron OIDC Adapter .............. SUCCESS [ 0.303 s]
[INFO] Keycloak Wildfly Adapter Subsystem ................. SUCCESS [ 3.558 s]
[INFO] Keycloak WildFly Integration ....................... SUCCESS [ 0.015 s]
[INFO] Keycloak OIDC Client Adapter Modules ............... SUCCESS [ 0.012 s]
[INFO] Keycloak Adapters .................................. SUCCESS [ 0.015 s]
[INFO] Keycloak Misc ...................................... SUCCESS [ 0.012 s]
[INFO] Keycloak :: Spring :: Boot ......................... SUCCESS [ 0.016 s]
[INFO] Keycloak :: Spring :: Boot :: Default :: Starter .. SUCCESS [ 15.382 s]
[INFO] keycloak-test-helper ............................... SUCCESS [ 25.080 s]
[INFO] Keycloak JavaScript Admin Client ................... SUCCESS [ 19.007 s]
[INFO] Keycloak JavaScript Adapter ........................ SUCCESS [ 4.279 s]
[INFO] Keycloak Quarkus Server App ........................ SUCCESS [ 37.902 s]
[INFO] Keycloak Quarkus Server Distribution ............... SUCCESS [02:49 min]
[INFO] Keycloak Quarkus Test Parent ....................... SUCCESS [ 0.010 s]
[INFO] Keycloak Quarkus Server JUnit 5 Internal Test Framework SUCCESS [ 22.687 s]
[INFO] Keycloak Quarkus Server Integration tests .......... FAILURE [ 01:37 h]
[INFO] Keycloak TestSuite ................................. SKIPPED
[INFO] DB Allocator Plugin ................................ SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite .......... SKIPPED
[INFO] Test apps .......................................... SKIPPED
[INFO] Test apps distribution ............................. SKIPPED
[INFO] Keycloak Authz Tests: Hello World Example .......... SKIPPED
[INFO] Keycloak Authz: Servlet Authorization Test ......... SKIPPED
[INFO] Keycloak Authz: Simple Servlet App with Policy Enforcer SKIPPED
[INFO] integration-arquillian-test-apps-servlets .......... SKIPPED
[INFO] Keycloak Test App Profile JEE ...................... SKIPPED
[INFO] integration-arquillian-test-apps-cors-parent ....... SKIPPED
[INFO] Angular Product Portal JS .......................... SKIPPED
[INFO] JAX-RS Database Service Using OAuth Bearer Tokens .. SKIPPED
[INFO] spring-boot-adapter-app ............................ SKIPPED
[INFO] Fuse Test Applications ............................. SKIPPED
[INFO] Customer Portal - Secured in Karaf/Fuse ............ SKIPPED
[INFO] CXF JAXRS Example - Secured in Karaf/Fuse .......... SKIPPED
[INFO] CXF JAXRS Example - Secured in Karaf/Fuse 7 on Undertow SKIPPED
[INFO] CXF JAXWS Example - Secured in Karaf/Fuse .......... SKIPPED
[INFO] CXF JAXWS Example - Secured in Karaf/Fuse 7 on Undertow SKIPPED
[INFO] Product Portal - Secured in Karaf/Fuse ............. SKIPPED
[INFO] Product Portal - Secured in Karaf/Fuse 7 on Undertow SKIPPED
[INFO] Camel endpoint example - Secured in Karaf/Fuse ..... SKIPPED
[INFO] Camel endpoint example - Secured in Karaf/Fuse 7.0 on Undertow SKIPPED
[INFO] Keycloak Fuse Example - Features ................... SKIPPED
[INFO] Keycloak Examples - External Config ................ SKIPPED
[INFO] Servers ............................................ SKIPPED
[INFO] Testsuite Adapter SPI .............................. SKIPPED
[INFO] Adapter Undertow SPI (JakartaEE) ................... SKIPPED
[INFO] Undertow OIDC Adapter (JakartaEE) .................. SKIPPED
[INFO] Undertow SAML Adapter (JakartaEE) .................. SKIPPED
[INFO] Auth Server ........................................ SKIPPED
[INFO] Auth Server Services ............................... SKIPPED
[INFO] Auth Server Services - Testsuite Providers Deployment SKIPPED
[INFO] Auth Server Services - Testsuite Providers ......... SKIPPED
[INFO] Keycloak TestSuite Utils ........................... SKIPPED
[INFO] Test Util .......................................... SKIPPED
[INFO] Auth Server - Undertow ............................. SKIPPED
[INFO] Auth Server - Quarkus .............................. SKIPPED
[INFO] App Server ......................................... SKIPPED
[INFO] App Server - SPI ................................... SKIPPED
[INFO] App Server - JBoss ................................. SKIPPED
[INFO] App Server - Wildfly ............................... SKIPPED
[INFO] App Server - Karaf ................................. SKIPPED
[INFO] App Server - Tomcat ................................ SKIPPED
[INFO] App Server - Undertow .............................. SKIPPED
[INFO] App Server - Jetty Parent .......................... SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Infinispan SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Legacy SKIPPED
[INFO] Tests .............................................. SKIPPED
[INFO] Base TestSuite ..................................... SKIPPED
[INFO] Other Tests Modules ................................ SKIPPED
[INFO] Tests for logical storage layer .................... SKIPPED
[INFO] Keycloak Examples .................................. SKIPPED
[INFO] Keycloak Examples - Admin Client ................... SKIPPED
[INFO] Provider Examples .................................. SKIPPED
[INFO] REST Example ....................................... SKIPPED
[INFO] Domain Extension Example ........................... SKIPPED
[INFO] Authenticator Example .............................. SKIPPED
[INFO] JS Console ......................................... SKIPPED
[INFO] Keycloak Examples - Kerberos Credential Delegation . SKIPPED
[INFO] Themes Examples .................................... SKIPPED
[INFO] SAML Examples ...................................... SKIPPED
[INFO] Keycloak SAML Adapter as a Servlet Filter .......... SKIPPED
[INFO] LDAP Demo Application .............................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:51 h
[INFO] Finished at: 2023-09-13T04:04:51+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7:test (default-test) on project keycloak-quarkus-integration-tests:
[ERROR]
[ERROR] Please refer to /home/pulkit/keycloak/quarkus/tests/integration/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR] mvn -rf :keycloak-quarkus-integration-tests

**Comment by crombus** [comment]
The image in quay for keycloak uses the dockerfile to build, my end goal to build that image.

**Comment by vmuzikar** [comment]
@crombus Can you please provide steps to reproduce, or even create a GH Issue for it? Thanks.

**Comment by crombus** [comment]
1) git clone <keycloak-repo>
2) install jdk17
3) install maven
4) run the command `mvn clean package -Doperator -Dquarkus.container-image.build=true `

**Comment by shawkins** [comment]
@crombus probably the simpliest thing to do initially is add -DskipTests just to confirm that it builds all the way.  From there if you are working only with the operator, then just change into the operator directory and don't bother with the rest of the project.

To determine what is happening with the Quarkus Server Integration tests will take seeing more of error that is occurring.  After running with -DskipTests, you could try going specifically into the quarkus server integration tests module and running just a single test "mvn clean package -Dtest=OptionValidationTest" and capture the log.

**Reply by crombus** [reply]
yes, that worked, thanks.
